internal/lang/funcs/collection.go

```diff
@@ -578,13 +580,22 @@ var TransposeFunc = function.New(&function.Spec{
 		outputMap := make(map[string]cty.Value)
 		tmpMap := make(map[string][]string)
 
+		path := make(cty.Path, 0, 2)
```

Comment:
We should just turn off mnd at this point

Reply:
Well, perhaps! But in this case I noticed the lint failure and already fixed it so I guess we can save that decision for later. :grinning:

Reply:
Would you be up for using the nolint directive instead so we can disable it this week?

Reply:
If I'm understanding correctly that you'd prefer this to just end up with an unnamed `2` as I originally wrote it then I'm happy to just sit on this for now and rebase and revise it once we've updated the lint rules, since this fix is not urgent.

Reply:
We'll be able to switch back to how I originally wrote this (with `2` directly inline as a "magic number") once https://github.com/opentofu/opentofu/pull/2577 is merged.

Reply:
I have rebased this onto the changes that #2577 made in `main` and reverted back to using just an inline magic number again as I'd originally wrote it. However, this time I did at least include a comment explaining why two is an appropriate value to use for capacity here.